pkg/controller/external_nofork.go

```diff
@@ -264,18 +278,53 @@ func (c *NoForkConnector) Connect(ctx context.Context, mg xpresource.Managed) (m
 	}
 
 	return &noForkExternal{
-		ts:             ts,
-		resourceSchema: c.config.TerraformResource,
-		config:         c.config,
-		params:         params,
-		rawConfig:      rawConfig,
-		logger:         logger,
-		metricRecorder: c.metricRecorder,
-		opTracker:      opTracker,
+		ts:                             ts,
+		resourceSchema:                 c.config.TerraformResource,
+		config:                         c.config,
+		params:                         params,
+		initProviderExclusiveParamKeys: initProviderExclusiveParams,
+		rawConfig:                      rawConfig,
+		logger:                         logger,
+		metricRecorder:                 c.metricRecorder,
+		opTracker:                      opTracker,
 	}, nil
 }
 
-func (n *noForkExternal) getResourceDataDiff(ctx context.Context, s *tf.InstanceState) (*tf.InstanceDiff, error) {
+func deleteInstanceDiffAttribute(instanceDiff *tf.InstanceDiff, paramKey string) error {
+	delete(instanceDiff.Attributes, paramKey)
+
+	keyComponents := strings.Split(paramKey, ".")
+	if len(keyComponents) < 2 {
+		return nil
+	}
+
+	keyComponents[len(keyComponents)-1] = "%"
+	lengthKey := strings.Join(keyComponents, ".")
+	if lengthValue, ok := instanceDiff.Attributes[lengthKey]; ok {
+		newValue, err := strconv.Atoi(lengthValue.New)
+		if err != nil {
+			return errors.Wrap(err, "cannot convert instance diff attribute to integer")
+		}
+
+		// TODO: consider what happens if oldValue = ""
+		oldValue, err := strconv.Atoi(lengthValue.Old)
+		if err != nil {
+			return errors.Wrap(err, "cannot convert instance diff attribute to integer")
+		}
+
+		newValue -= 1
+		if oldValue == newValue {
+			delete(instanceDiff.Attributes, lengthKey)
```

Comment:
Shouldn't we decide whether to delete the length key or not after we process (filter or not) all the keys?

Reply:
Yes, doing so is a better approach. So, we processed length attributes separately in the [next PR](https://github.com/ulucinar/upbound-upjet/pull/6).